src/validators/helpers.js

```diff
@@ -311,3 +311,32 @@ export const buildDate = date => {
     return null
   }
 }
+
+export const pickDate = (dates, max=true) => {
+ const buildsDates = dates.map(buildDate)
+
+ const findMinDate = (finalDate, dateItem) => {
+   if (finalDate <= dateItem) {
+     return finalDate
+   }
+   else {
+     return dateItem
+   }
+ }
+
+ const findMaxDate = (finalDate, dateItem) => {
+   if (finalDate >= dateItem) {
+     return finalDate
+   }
+   else {
+     return dateItem
+   }
+ }
+
+ if (max) {
+    return buildsDates.reduce(findMaxDate)
+  }
+  else {
```

Comment:
Since we have a return here, the `else` block is unnecessary. You can do `return buildsDates.reduce(findMinDate)`